sstables/sstables.hh

```diff
@@ -220,6 +221,12 @@ public:
     // will move it into a quarantine_dir subdirectory of its current directory.
     future<> move_to_quarantine(delayed_commit_changes* delay = nullptr);
 
+    // Move the sstable between states
+    //
+    // Known states are normal, staging, upload and quarantine.
+    // It's up to the storage driver how to implement this.
+    future<> change_state(sstring to, delayed_commit_changes* delay = nullptr);
```

Comment:
Would love for `to` to be an `enum class` instead of a string.

Reply:
Absolutely. Once I finish #12707 , it _will_ be an `enum class`